front/lib/api/assistant/conversation.ts

```diff
@@ -415,7 +440,16 @@ export async function getConversation(
           });
           return { m, rank: message.rank, version: message.version };
         }
-        throw new Error("Unreachable: message must be either user or agent");
+        if (message.contentFragment) {
```

Comment:
frustrating, but with this setup we cannot use the type system to test for exhaustivity.